### ISSUE TITLE
FASTQ input format needs to support index sequences

### ISSUE THREAD

**Issue by fnothaft** [bigdatagenomics/adam#1697]
Related to #1585. There's another Illumina FASTQ spec out there that puts the index sequences in the read name. From the ["home of the FASTQ spec"](https://en.wikipedia.org/wiki/FASTQ_format#Illumina_sequence_identifiers):

```
@EAS139:136:FC706VJ:2:2104:15343:197393 1:Y:18:ATCACG
```

I've also seen something like this for paired sequences:

```
@EAS139:136:FC706VJ:2:2104:15343:197393 1:Y:18:ATCACG+ATCACG
```

This is close to but slightly different from the variant in #1585.

**Comment by heuermh** [comment]
The "spec" as it were is [this paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2847217/), what you are describing is more stuff-whatever-the-hell-in-the-description-line, which has been an ongoing problem with FASTA format for like 20 years.  😄

**Comment by heuermh** [comment]
> for like 20 years.

Shit, I just realized I've been doing this for more than 20 years already.  And FASTA the tool was published in 1985.

**Comment by heuermh** [comment]
From [gitter](https://gitter.im/bigdatagenomics/adam?at=5b32a2c37d3bca737a0c8136), a pair of reads with index sequences:
```
@HISEQ_HU01:89:H7YRLADXX:1:1101:1116:2123 1:N:0:ATCACG
@HISEQ_HU01:89:H7YRLADXX:1:1101:1116:2123 2:N:0:ATCACG
```

**Comment by heuermh** [comment]
Fixed by #2004

**Comment by heuermh** [comment]
Reopening per https://github.com/bigdatagenomics/adam/commit/e7cbfcd2ae32f41d38e9dfdbd295f593d2f7f199#r29602988
